raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -211,6 +216,70 @@ contract MonitoringService is Utils {
         );
     }
 
+    function isAllowedToMonitor(
+        TokenNetwork token_network,
+        uint256 channel_identifier,
+        address closing_participant,
+        address non_closing_participant,
+        address monitoring_service_address
+    )
+        internal
+        returns (bool)
+    {
+        TokenNetwork.ChannelState channel_state;
+        uint256 settle_block_number;
+        (settle_block_number, channel_state) = token_network.getChannelInfo(
+            channel_identifier, closing_participant, non_closing_participant
+        );
+        require(channel_state == TokenNetwork.ChannelState.Closed);
+
+        // We don't actually know when the channel has been closed. So we'll
+        // make a guess so that assumed_close_block >= real_close_block.
+        uint256 assumed_settle_timeout = token_network.settlement_timeout_min();
+        uint256 assumed_close_block = settle_block_number - assumed_settle_timeout;
```

Comment:
Before the subtraction, let's add `require(settle_block_number >= assumed_settle_timeout);`.

`token_network` is an externally given address and can be arbitrarily nasty.